vscode-dotnet-runtime-library/src/test/unit/LinuxVersionResolver.test.ts

```diff
@@ -0,0 +1,159 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+import * as chai from 'chai';
+import * as os from 'os';
+import { GenericDistroSDKProvider } from '../../Acquisition/GenericDistroSDKProvider';
+import { MockCommandExecutor, MockDistroProvider, MockEventStream, MockExtensionContext, MockInstallationValidator, NoInstallAcquisitionInvoker } from '../mocks/MockObjects';
+import { DistroVersionPair, DotnetDistroSupportStatus, LinuxVersionResolver } from '../../Acquisition/LinuxVersionResolver';
+import { IAcquisitionWorkerContext } from '../../Acquisition/IAcquisitionWorkerContext';
+import { RuntimeInstallationDirectoryProvider } from '../../Acquisition/RuntimeInstallationDirectoryProvider';
+import { SdkInstallationDirectoryProvider } from '../../Acquisition/SdkInstallationDirectoryProvider';
+const assert = chai.assert;
+const standardTimeoutTime = 100000;
+
+
+function mockContext(runtimeInstall: boolean): IAcquisitionWorkerContext {
+    const extensionContext = new MockExtensionContext();
+    const eventStream = new MockEventStream();
+    const workerContext : IAcquisitionWorkerContext = {
+        storagePath: '',
+        extensionState: extensionContext,
+        eventStream,
+        acquisitionInvoker: new NoInstallAcquisitionInvoker(eventStream),
+        installationValidator: new MockInstallationValidator(eventStream),
+        timeoutValue: standardTimeoutTime,
+        installDirectoryProvider: runtimeInstall ? new RuntimeInstallationDirectoryProvider('') : new SdkInstallationDirectoryProvider(''),
+    };
+    return workerContext;
+}
+
+suite('Linux Version Resolver Tests', () =>
+{
```

Comment:
I *think* it might be possible to include a conditional here and wrap all the tests in an if statement, so you don't have to check shouldRun in each individually: https://stackoverflow.com/questions/17672825/conditional-execution-of-mocha-test-cases

Reply:
That makes things easier, thanks :)

Reply:
Ah, it actually just never runs the tests now. Sadly I have to undo this change. Wish we could keep it like this,  it was a really good suggestion!